core/src/main/java/io/grpc/internal/InternalSubchannel.java

```diff
@@ -345,8 +345,9 @@ private void gotoState(final ConnectivityStateInfo newState) {
     syncContext.throwIfNotInThisSynchronizationContext();
 
     if (state.getState() != newState.getState()) {
-      Preconditions.checkState(state.getState() != SHUTDOWN,
-          "Cannot transition out of SHUTDOWN to " + newState);
```

Comment:
The conventional way to solve this is to use the message template:
```java
      Preconditions.checkState(state.getState() != SHUTDOWN,
          "Cannot transition out of SHUTDOWN to %s", newState);
```

In addition, we can also consider using `newState.getState()` in the error message, as it looks like a bug that the full state info was included.